tools/clang/lib/SPIRV/DebugTypeVisitor.cpp

```diff
@@ -260,12 +303,28 @@ bool DebugTypeVisitor::visitInstruction(SpirvInstruction *instr) {
     // DebugGlobalVariable
     // DebugLocalVariable
     // DebugFunction
-    // DebugFunctionDeclaration
-    // TODO: We currently don't have a SpirvDebugFunctionDeclaration class. Add
-    // one if needed.
     if (isa<SpirvDebugGlobalVariable>(debugInstr) ||
         isa<SpirvDebugLocalVariable>(debugInstr)) {
       const SpirvType *spirvType = debugInstr->getDebugSpirvType();
+      if (!spirvType) {
+        // If a debug instruction does not have QualType, LowerTypeVisitor
+        // does not invoke lowerDebugTypeComposite() even though the type is a
+        // composite type. For example, cbuffer does not have QualType, but it
+        // has StructType. We have to prepare a DebugTypeComposite for such
+        // StructType before calling
+        // DebugTypeVisitor::lowerToDebugTypeComposite().
+        if (auto *debugGlobalVar = dyn_cast<SpirvDebugGlobalVariable>(instr)) {
+          auto *varType = debugGlobalVar->getVariable()->getResultType();
+          assert(varType &&
+                 "Global variables must be lowered by LowerTypeVisitor");
+          auto *ptrType = dyn_cast<SpirvPointerType>(varType);
+          assert(ptrType && "OpVariable must have a pointer type");
+          spirvType = ptrType->getPointeeType();
+          debugGlobalVar->setDebugSpirvType(spirvType);
+          if (auto *structType = dyn_cast<StructType>(spirvType))
+            prepareDebugTypeComposite(structType, instr->getSourceLocation());
```

Comment:
It would be better to call this `lowerCbufferDebugType` so that it doesn't get mixed up with regular structures

Reply:
done